src/main/java/com/google/devtools/build/lib/skyframe/TreeArtifactValue.java

```diff
@@ -233,6 +245,11 @@ public Optional<ArchivedRepresentation> getArchivedRepresentation() {
     return Optional.ofNullable(archivedRepresentation);
   }
 
+  /** Return symlinkTarget path (if present). */
+  public Optional<PathFragment> getSymlinkTargetExecPath() {
```

Comment:
As we talked offline, you may be better off returning a FileMetadataValue from `getMetadata` which includes that so you can just request that using `metadataProvider.getMetadata(treeArtifact)`.

Reply:
Thanks, this made everything so much simpler!